### DISCUSSION TITLE
Does the auto-scaling consider available hardware resources?

### DISCUSSION THREAD

**Discussion by Martiix** [actions/actions-runner-controller#1082] [Questions] [answered]
Hi, I have been looking into optimizing utilization of the available hardware for continuous integration and this solution seems promising. However, I get the impression that the max and min replicas of runners are hardcoded? Does this system for instance take available CPU and memory of the underlying hardware into account when scaling up/down?

**Comment by cbui** [comment, marked as answer]
> However, I get the impression that the max and min replicas of runners are hardcoded?

There's a `desiredReplicas` field that determines the actual number of replicas running, bounded by the `min` and `max`. 

> Does this system for instance take available CPU and memory of the underlying hardware into account when scaling up/down?

If you mean the available cluster resources, no. You'll want to use a solution such as https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler

**Reply by Martiix** [reply]
Thank you for the reply, this is very helpful.
